apps/website/src/pages/hackathon/[id].astro

```diff
@@ -2,346 +2,310 @@
-		},
-		{
-			title: '#5: Publish the source code',
-			description:
-				'Release your app’s code on GitHub or a similar open source platform.',
-		},
-		{
-			title: '#6: Submit before 11:59pm @ August 18, 2025',
-			description:
-				'Submit your app before the deadline to qualify for rewards.',
-		},
-	],
-	faqs: [
-		{
-			title: 'Are teams allowed?',
-			description: [
-				'Yes! While building solo is absolutely fine, we encourage teams of 2 for these challenges. Part of the fun is collaborating with other people.',
-				'However, please note that while teams of any size are allowed, each project will receive a maximum of 2 of each monetary or product reward.',
-			],
-		},
-		{
-			title: 'Am I allowed to use AI?',
-			description: [
-				'You’re allowed to use any tool that you would use as part of your regular workflow. We only ask that you try to keep in the spirit of the challenge, which is to learn something new.',
-				'We do reserve the right to deny rewards to any submissions that appear to be entirely generated by AI, as that violates the spirit of the challenge.',
-			],
-		},
-		{
-			title:
-				'I’m not sure I have enough experience to build this. Should I still try?',
-			description: [
-				'Yes! These challenges are perfect for trying something new with a limited time commitment and low stakes. The only way to learn stuff is to do stuff!',
-				'This is also a great opportunity to pair with someone and help each other level up. Ask a friend, or hit up the CodeTV Discord to find someone to work with.',
-			],
-		},
-	],
-};
+export async function getStaticPaths() {
+	const hackathons = await getAllHackathons();
+
+	return hackathons.map((hackathon) => ({
+		params: { id: hackathon.slug },
+		props: { slug: hackathon.slug },
+	}));
+}
+
+const { slug } = Astro.props;
+const hackathon = await getHackathonBySlug({ slug: slug! });
+
+if (!hackathon) {
+	return Astro.redirect('/404');
+}
+
+// Calculate status based on deadline
+const now = new Date();
+const deadline = hackathon.deadline ? new Date(hackathon.deadline) : null;
+const isActive = deadline ? now < deadline : false;
+const statusText = isActive ? 'Active' : 'Closed';
+const statusClass = isActive ? 'active' : 'closed';
+
+// Format deadline for display
+const deadlineFormatted = deadline
+	? deadline.toLocaleDateString('en-US', {
+			month: 'short',
+			day: 'numeric',
+		})
+	: 'TBD';
+
+// Get first sponsor if available
+const primarySponsor = hackathon.sponsors?.[0];
+
+// Hero image - use hero_image, fall back to share_image, then default
+const heroImage = hackathon.hero_image
+	? {
+			public_id: hackathon.hero_image.public_id!,
+			alt: hackathon.title || 'Hackathon',
+		}
+	: hackathon.share_image
+		? {
+				public_id: hackathon.share_image.public_id!,
+				alt: hackathon.title || 'Hackathon',
+			}
+		: {
+				public_id: 'codetv-home-hero',
+				alt: 'CodeTV Web Dev Challenge',
+			};
```

Comment:
I generally prefer avoiding these kinds of nested ternaries since they're harder to quickly parse

maybe something like:

```js
let public_id = hackathon.hero_image ? hackathon.hero_image.public_id! : hackathon.share_image.public_id!;

const heroImage = {
				public_id: public_id || 'codetv-home-hero',
				alt: hackathon.title || 'Web Dev Challenge Hackathon',
};
```

Reply:
Good call. That reads a lot nicer